### PR TITLE
fix: addMinter setting wrong address + add tests

### LINKED ISSUE #14: `addMinter` function sets the sender as minter, not the target account
In `addMinter` [line 156](https://github.com/ApeAcademy/ERC20/blob/3b1cac5fbbcda904e08a7e47b980f281c830b828/%7B%7Bcookiecutter.project_name%7D%7D/contracts/Token.vy#L156), I think there is an error as to which account becomes a minter.

Shouldn't `minter` become minter? i.e.:

```python
@external
def addMinter(minter: address) -> bool:
    assert msg.sender == self.owner
    self.isMinter[minter] = True   # This line changes
    return True  # Also we probably want to add a return value to make this gas efficient
```

### PR DESCRIPTION
**Pull request by merc1er**
Fixes #14

### CODE REVIEW

**Comment by merc1er** [comment]
Hmm, would you know why I am getting:

```python
AttributeError: 'Token' has no attribute 'addMinter'.
```

@fubuloubu

**Comment by merc1er** [comment]
> Hmm, would you know why I am getting:
> 
> ```python
> AttributeError: 'Token' has no attribute 'addMinter'.
> ```
> 
> @fubuloubu

Got it, I was not familiar with `cookiecutter` and this workflow matrix. You might want to squash those commits if you merge this 🙂

File: {{cookiecutter.project_name}}/tests/test_token.py [hunk 1]
```diff
@@ -185,11 +185,33 @@ def test_mint(token, owner, receiver):
...
     totalSupply = token.totalSupply()
     assert totalSupply == 1420
+
+
+def test_add_minter(token, owner, accounts):
+    """
+    Test adding new minter.
+    Must trigger MinterAdded Event.
+    Must return true when checking if target isMinter
+    """
```
Comment (fubuloubu) [inline {{cookiecutter.project_name}}/tests/test_token.py:200]: I usually like to assert the pre condition too, e.g. That it's not set before you set it

File: {{cookiecutter.project_name}}/tests/test_token.py [hunk 2]
```diff
@@ -174,7 +174,7 @@ def test_mint(token, owner, receiver):
     totalSupply = token.totalSupply()
     assert totalSupply == 1000
 
-    receiver_balance = token.balanceOf(receiver) 
+    receiver_balance = token.balanceOf(receiver)
```
Comment (fubuloubu) [inline {{cookiecutter.project_name}}/tests/test_token.py:177]: What are all these small changes doing? I can't see the difference

**Review by fubuloubu — APPROVED** [review]

File: {{cookiecutter.project_name}}/tests/test_token.py (hunk 2 above)
Comment (merc1er) [reply {{cookiecutter.project_name}}/tests/test_token.py:177]: There were extra white spaces at the end of those lines. You should be able to see the change in GitHub or in your text editor if you configure it to show every character.

image

Comment (fubuloubu) [reply {{cookiecutter.project_name}}/tests/test_token.py:177]: okay, that's what I figured. it doesn't show up on mobile github

File: {{cookiecutter.project_name}}/tests/test_token.py (hunk 1 above)
Comment (fubuloubu) [reply {{cookiecutter.project_name}}/tests/test_token.py:200]:
```suggestion
    """
    assert token.isMinter(target) is False
```

**Review by fubuloubu — DISMISSED** [review]

**Review by fubuloubu — DISMISSED** [review]

File: {{cookiecutter.project_name}}/tests/test_token.py [hunk 3]
```diff
@@ -185,11 +185,34 @@ def test_mint(token, owner, receiver):
...
+
+
+def test_add_minter(token, owner, accounts):
+    """
+    Test adding new minter.
+    Must trigger MinterAdded Event.
+    Must return true when checking if target isMinter
+    """
+    assert token.isMinter(target) is False
+    target = accounts[1]
```
Comment (fubuloubu) [inline {{cookiecutter.project_name}}/tests/test_token.py]:
```suggestion
    target = accounts[1]
    assert token.isMinter(target) is False
```

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**